tests/LoggerManager.spec.ts

```diff
@@ -0,0 +1,150 @@
+			expect(client.listenerCount(event)).toBe(modules.length);
+		});
+
+		it('should not add twice an event to the events list', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+
+			sut.listenWithAllModulesTo(event);
+			sut.listenWithAllModulesTo(event);
+
+			expect(client.listenerCount(event)).toBe(modules.length);
+		});
+
+		it('should not add event if requirements are not met', () => {
+			client = new Client({ intents: [] });
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+
+			sut.listenWithAllModulesTo(event);
+
+			expect(client.listenerCount(event)).toBe(0);
+		});
+	});
+
+	describe('unlistenWithAllModulesTo', () => {
+		it('should remove event from the events list', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+
+			sut.listenWithAllModulesTo(event);
+			sut.unlistenWithAllModulesTo(event);
+
+			expect(client.listenerCount(event)).toBe(0);
+		});
+
+		it('should not remove any event from the events list if the event is not registered in the first place', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+
+			expect(client.listenerCount(event)).toBe(0);
+			sut.unlistenWithAllModulesTo(event);
+			expect(client.listenerCount(event)).toBe(0);
+		});
+	});
+
+	describe('listenWithModuleTo', () => {
+		it('should listen to event with the right module', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+			const module = { [`on_${event}`]: () => { } };
+
+			expect(client.listenerCount(event)).toBe(0);
+			sut.listenWithModuleTo(event, module);
+			expect(client.listenerCount(event)).toBe(1);
+		});
+
+		it('should not add twice the same module to listen to an event', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+			const module = { [`on_${event}`]: () => { } };
+
+			expect(client.listenerCount(event)).toBe(0);
+			sut.listenWithModuleTo(event, module);
+			sut.listenWithModuleTo(event, module);
+			expect(client.listenerCount(event)).toBe(1);
+		});
+
+		it('should listen to an event with different modules', () => {
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+			const module = { [`on_${event}`]: () => { } };
+			const moduleTwo = { [`on_${event}`]: () => { } };
+
+			expect(client.listenerCount(event)).toBe(0);
+			sut.listenWithModuleTo(event, module);
+			sut.listenWithModuleTo(event, moduleTwo);
+			expect(client.listenerCount(event)).toBe(2);
+		});
+
+		it('should not add event if requirements are not met', () => {
+			client = new Client({ intents: [] });
+			const sut = new LoggerManager(client, modules);
+			const event = getTestEvent();
+			const module = { [`on_${event}`]: () => { } };
+
+			sut.listenWithModuleTo(event, module);
+
+			expect(client.listenerCount(event)).toBe(0);
+		});
```

Comment:
Should be removed if the intents check and the event parser are not mandatory anymore